src/wattsi.pas

```diff
@@ -158,6 +165,90 @@    TCrossReferences = record
    CurrentSectionNumber: array[THeadingRank] of Cardinal;
    StringStore: TStringStore;
    Errors: Cardinal;
+   XrefsByDFNAnchor: TXrefsByDFNAnchor;
+
+   function NewXrefsByDFNAnchor(SectionName: UTF8String; Anchor: UTF8String): TXrefAnchorsBySectionName;
+   var
+      Anchors: TStringList;
+      XrefAnchorsBySectionName: TXrefAnchorsBySectionName;
+   begin
+      Anchors := TStringList.Create;
+      Anchors.Add(Anchor);
+      XrefAnchorsBySectionName := TXrefAnchorsBySectionName.Create;
+      XrefAnchorsBySectionName.Add(SectionName, Anchors);
+      Result := XrefAnchorsBySectionName;
+   end;
+
+   function NewXrefAnchorsBySectionName(SectionName: UTF8String; Anchor: UTF8String): TXrefAnchorsBySectionName;
+   var
+      Anchors: TStringList;
+      XrefAnchorsBySectionName: TXrefAnchorsBySectionName;
+   begin
+      Anchors := TStringList.Create;
+      Anchors.Add(Anchor);
+      XrefAnchorsBySectionName := TXrefAnchorsBySectionName.Create;
+      XrefAnchorsBySectionName.Add(SectionName, Anchors);
+      Result := XrefAnchorsBySectionName;
+   end;
+
+   procedure XrefsToJSON(XrefsByDFNAnchor: TXrefsByDFNAnchor);
```

Comment:
FWIW it looks like there might be a way to use a package for this: http://wiki.freepascal.org/fcl-json. However I'm not sure it's worth the effort, especially if adding a dependency is nontrivial.